cmd/akamai-purger/main.go

```diff
@@ -228,7 +228,7 @@ func (ap *akamaiPurger) Purge(ctx context.Context, req *akamaipb.PurgeRequest) (
 }
 
 func main() {
-	daemonFlags := flag.NewFlagSet("daemon", flag.ExitOnError)
+	daemonFlags := flag.NewFlagSet("daemon", flag.ContinueOnError)
```

Comment:
Should line 236 be updated to have the same?

Reply:
No - the goal of ContinueOnError is that we get a chance to manually handle errors by printing help for the "manual" subcommand.

The flagset on line 236 only gets applied if we've already seen a "manual" subcommand; in that case I don't think we need to provide help about the daemon mode, since the user is specifically interested in manual mode.

Reply:
Ahhh, I'd missed the bit where this was being executed in a bare `else`, so this is the default code-path.

But that means that
a) if someone does run it _explicitly_ in daemon mode, but gets the flags wrong, they'll be getting help text for manual mode too, which feels a bit odd; and
b) if someone runs `go run akamai-purge foobleach -config ...` then this'll default into daemon mode without even batting an eye, which feels wrong (but admittedly isn't the fault of this PR).

So maybe it would be best to make this a four-way case: explicitly manual, explicitly daemon, explicitly something unrecognized, or nothing specified (default to daemon, but print both sets of help if the flags don't parse).

But maybe that's just more than you want to tackle in this PR?